src/main/java/org/cru/redegg/reporting/InMemoryErrorQueue.java

```diff
@@ -82,6 +93,22 @@ public void shutdown()
     @Override
     public void enqueue(final ErrorReport report)
     {
+        try
+        {
+            submit(report);
+        }
+        catch (RejectedExecutionException e)
+        {
+            System.out.println("submission rejected");
+            errorLog.error("unable to submit error report to queue; using fallback reporter", e);
+
+            // run on this thread
+            fallback(report);
+        }
+    }
+
+    private void submit(final ErrorReport report)
```

Comment:
What do you think of https://github.com/hlbraddock/red-egg/commit/443c16580cd954ee95ee2be0ab6b5942d36e317e ?